unixfs/examples/get.rs

```diff
@@ -70,49 +70,44 @@ fn walk(blocks: ShardedBlockStore, start: &Cid) -> Result<(), Error> {
-                // presense of metadata can be used to determine if this is the first apperiance of
-                // a directory by looking at the metadata: sibling hamt shard buckets do not have
-                // metadata.
-                if let Entry::Metadata(metadata_entry) = item.as_entry() {
-                    let metadata = metadata_entry.metadata();
-                    let path = item.as_entry().path();
-                    let mode = metadata.mode().unwrap_or(0o0755) & 0o7777;
-                    let (seconds, _) = metadata.mtime().unwrap_or((0, 0));
-                    println!("d {:o} {:>12} {:>16} {:?}", mode, seconds, "-", path);
+
+                if segment.is_last() {
+                    let mode = md.mode().unwrap_or(0o0644) & 0o7777;
+                    let (seconds, _) = md.mtime().unwrap_or((0, 0));
+                    println!("f {:o} {:>12} {:>16} {:?}", mode, seconds, size, path);
                 }
-                item.into_inner()
             }
-            ContinuedWalk::Symlink(bytes, item) => {
-                if let Entry::Metadata(metadata_entry) = item.as_entry() {
-                    let metadata = metadata_entry.metadata();
-                    let path = metadata_entry.path();
-                    let target = Path::new(std::str::from_utf8(bytes).unwrap());
-                    let mode = metadata.mode().unwrap_or(0o0755) & 0o7777;
-                    let (seconds, _) = metadata.mtime().unwrap_or((0, 0));
-                    println!(
-                        "s {:o} {:>12} {:>16} {:?} -> {:?}",
-                        mode, seconds, "-", path, target
-                    );
-                }
-                item.into_inner()
+            // presense of metadata can be used to determine if this is the first apperiance of
+            // a directory by looking at the metadata: sibling hamt shard buckets do not have
+            // metadata.
+            ContinuedWalk::Directory(Entry::Metadata(metadata_entry)) => {
+                let metadata = metadata_entry.metadata();
+                let path = metadata_entry.path();
+                let mode = metadata.mode().unwrap_or(0o0755) & 0o7777;
+                let (seconds, _) = metadata.mtime().unwrap_or((0, 0));
+                println!("d {:o} {:>12} {:>16} {:?}", mode, seconds, "-", path);
+            }
+            ContinuedWalk::Symlink(bytes, Entry::Metadata(metadata_entry)) => {
+                let metadata = metadata_entry.metadata();
+                let path = metadata_entry.path();
+                let target = Path::new(std::str::from_utf8(bytes).unwrap());
+                let mode = metadata.mode().unwrap_or(0o0755) & 0o7777;
+                let (seconds, _) = metadata.mtime().unwrap_or((0, 0));
+                println!(
+                    "s {:o} {:>12} {:>16} {:?} -> {:?}",
+                    mode, seconds, "-", path, target
+                );
             }
+            _ => {}
```

Comment:
I think with my suggestions above this would now be an error.
```suggestion
```

Reply:
If this suggestion is followed it leaves the `File(_, Bucket(_, _))` unhandled. I don't think this variant should be possible, due to how the "continuations" carry over the metadata for files but not for buckets.

Reply:
I see now. Thanks for clarifying it

Reply:
Oki, looking at this now, resolving other comment branches. This can be resolved, I'll try continue a proper review or a PR against your code.